src/Symfony/Component/Console/Tests/ApplicationTest.php

```diff
@@ -914,7 +914,7 @@ public function testRunWithDispatcher()
     }
 
     /**
-     * @expectedException        \LogicException
+     * @expectedException        \RuntimeException
```

Comment:
So my bad, this can't change, that would be a bc break.
I propose to tweak the implementation like this, so that the original exception is preserved, except when a listener decides to alter it:
```php
            try {
                $e = null;
                $exitCode = $command->run($input, $output);
            } catch (\Exception $x) {
                $e = $x;
            } catch (\Throwable $x) {
                $e = new FatalThrowableError($x);
            }
            if (null !== $e) {
                $event = new ConsoleExceptionEvent($command, $input, $output, $e, $e->getCode());
                $this->dispatcher->dispatch(ConsoleEvents::EXCEPTION, $event);

                if ($e !== $event->getException()) {
                    $x = $e = $event->getException();
                }

                $event = new ConsoleTerminateEvent($command, $input, $output, $e->getCode());
                $this->dispatcher->dispatch(ConsoleEvents::TERMINATE, $event);

                throw $x;
            }
```